src/type_object.rs

```diff
@@ -86,9 +86,6 @@ pub unsafe trait PyTypeInfo: Sized {
 
 /// Implemented by types which can be used as a concrete Python type inside `Py<T>` smart pointers.
 pub trait PyTypeCheck {
-    /// Name of self. This is used in error messages, for example.
-    const NAME: &'static str;
```

Comment:
Can we make this `#[deprecated]` instead of removing? The deprecation message can recommend users get a "name" at runtime from `classinfo_object`, perhaps.

(We might be able to default it to empty, so that users don't need to implement it, we _might_ want to leave existing implementations in PyO3 for a couple of versions just in case users are depending on it.)